lib/src/presentation/bloc/movies_bloc.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:async';
+import '../../core/util/data_state.dart';
+import '../../domain/entity/app_event.dart';
+import '../../domain/entity/movie_categories.dart';
+import '../../domain/usecase/usecase_interface.dart';
+import '../../core/bloc/i_bloc.dart';
+
+class MoviesBloc extends IBloc {
+  IUseCase popularMoviesUseCase;
+  IUseCase topRatedMoviesUseCase;
+  IUseCase nowPlayingMoviesUseCase;
+  IUseCase upcomingMoviesUseCase;
+
+  MoviesBloc({
+    required this.popularMoviesUseCase,
+    required this.topRatedMoviesUseCase,
+    required this.nowPlayingMoviesUseCase,
+    required this.upcomingMoviesUseCase,
+  });
+
+  final _moviesController = StreamController<AppEvent>.broadcast();
+
+  Stream<AppEvent> get allMovies => _moviesController.stream;
+
+  @override
+  void dispose() {
+    _moviesController.close();
+  }
+
+  @override
+  void initialize() {
+    // TODO: implement initialize
+  }
+
+  void fetchAllMovies() async {
+    DataState popularMoviesFromApi = await popularMoviesUseCase.call();
+    DataState topRatedMoviesFromApi = await topRatedMoviesUseCase.call();
+    DataState nowPlayingMoviesFromApi = await nowPlayingMoviesUseCase.call();
+    DataState upcomingMoviesFromApi = await upcomingMoviesUseCase.call();
+
+    if (popularMoviesFromApi is DataSuccess &&
+        topRatedMoviesFromApi is DataSuccess &&
+        nowPlayingMoviesFromApi is DataSuccess &&
+        upcomingMoviesFromApi is DataSuccess) {
```

Comment:
Why you are calling four services at almost the same time?

Reply:
Because i'm showing the four categories on the home screen simultaneously, and each use case is initialized with its own endpoint, i'll try to find a cleaner way to do this

Reply:
4 streams :)

Reply:
You're right guys, this isn't scalable and performant at all, i'm refactoring it